### PR TITLE
Games

### PR DESCRIPTION
**Pull request by rekahamburger**
feladat megoldás- 1 ső probálkozás

### CODE REVIEW

File: src/BaseGame.ts [hunk 1]
```diff
@@ -22,15 +22,31 @@ export default abstract class BaseGame implements IBaseGame {
      * majd kerekítsd lefelé, ez lesz az index.
      * Majd térj vissza a this.levels tömbnek ezzel az indexével.
      */
-    abstract getRandomLevel(): Level;
+     abstract getRandomLevel(levels: any[]): Level {
```
Comment (bankiszabolcs) [inline src/BaseGame.ts:25]: protected head: Piece;
  protected tail: Piece;
  protected paused: boolean = false;
  protected food: Piece | null;
  protected goldenApple: Piece | null;
  protected length: number = 0;
  protected growth: number = 0;
  protected score: number = 0;
  protected currentLevel: Level | null;
  protected garden: HTMLDivElement;

File: src/BaseGame.ts [hunk 2]
```diff
@@ -22,15 +22,31 @@ export default abstract class BaseGame implements IBaseGame {
...
+        let feladat = Math.random();
+        this.levels= new Level;
+        let rand = Math.floor(feladat*this.levels.length);
+         return rand;
+    }
+
```
Comment (bankiszabolcs) [inline src/BaseGame.ts:41]: protected head: Piece;
protected tail: Piece;
protected paused: boolean = false;
protected food: Piece | null;
protected goldenApple: Piece | null;
protected length: number = 0;
protected growth: number = 0;
protected score: number = 0;
protected currentLevel: Level | null;
protected garden: HTMLDivElement;

File: src/BaseGame.ts [hunk 3]
```diff
@@ -22,15 +22,31 @@ export default abstract class BaseGame implements IBaseGame {
      * majd kerekítsd lefelé, ez lesz az index.
      * Majd térj vissza a this.levels tömbnek ezzel az indexével.
      */
-    abstract getRandomLevel(): Level;
+     abstract getRandomLevel(levels: any[]): Level {
+        let feladat = Math.random();
+        this.levels= new Level;
+        let rand = Math.floor(feladat*this.levels.length);
+         return rand;
```
Comment (bankiszabolcs) [inline src/BaseGame.ts]: nem kell paraméter
let helyett const
27-es sor nem kell
return this.levels[rand]

File: src/BaseGame.ts [hunk 4]
```diff
@@ -22,15 +22,31 @@ export default abstract class BaseGame implements IBaseGame {
...
+
 
     /**
      * @returns {boolean}
      * 1. hozz létre egy chance nevű változót 5 értékkel
      * 2. hozz létre egy pick nevű változót, értéke random szám szorozva 100-zal
      * 3. térj vissza true értékkel, ha a pick kisebb, mint a chance
      */
-    abstract mayIHaveGoldenApple(): boolean;
+     abstract mayIHaveGoldenApple(chance:number, pick:number): boolean{
```
Comment (bankiszabolcs) [inline src/BaseGame.ts:39]: paraméter!?

File: src/BaseGame.ts [hunk 5]
```diff
@@ -22,15 +22,31 @@ export default abstract class BaseGame implements IBaseGame {
...
+        chance= 5;
+        pick = Math.random ()*100;
+       if (pick<chance){
+           return true;
+       }else {
+           return false;
+       }
+   }
+
+
```
Comment (bankiszabolcs) [inline src/BaseGame.ts:59]: paraméter?

File: src/BaseGame.ts [hunk 6]
```diff
@@ -42,5 +58,14 @@ export default abstract class BaseGame implements IBaseGame {
...
+     abstract removeGrid (): void{
+        let grids = [];
+        grids.push(document.getElementsByClassName("vertical-grid horizontal-grid"));
+        for (let i=0; i<grids.length; i++ ){
+            Utils.removeNode(grids[i]); 
+        
+        }
+        this.gridVisible =false; 
+
+    }
```
Comment (bankiszabolcs) [inline src/BaseGame.ts:80]: Példa:
const grids: [] = Array.from(document.getElementsByClassName("vertical-grid horizontal-grid"))
VAGY: document.querySelectorAll

**Review by bankiszabolcs — CHANGES_REQUESTED** [review]
ASD

**Review by tamaslud — APPROVED** [review]
Kösz, ezekkel a módosításokkal szerintem jó lesz!

**Pull request merged**